python/flyft/state.py

```diff
@@ -47,44 +47,45 @@ def __getitem__(self, key):
         return self._cache[key]
 
 class Mesh(mirror.Mirror,mirrorclass=_flyft.Mesh):
-    def __init__(self, L, shape, origin=0.):
-        super().__init__(L, shape, origin)
 
     @property
-    def coordinates(self):
-        if not hasattr(self, '_coordinates'):
-            self._coordinates = np.array([self._self.coordinate(i) for i in range(self.shape)])
-        return self._coordinates
+    def centers(self):
+        if not hasattr(self, '_centers'):
+            self._centers = np.array([self._self.center(i) for i in range(self.shape)])
+        return self._centers
 
     L = mirror.Property()
     origin = mirror.Property()
     shape = mirror.Property()
     step = mirror.Property()
 
+class CartesianMesh(mirror.Mirror,mirrorclass=_flyft.CartesianMesh):
+    def __init__(self,L,shape,area = 1.):      
+        super().__init__(L,shape,area) 
+
+class SphericalMesh(mirror.Mirror,mirrorclass=_flyft.SphericalMesh):
+    def __init__(self,R,shape):
+        super().__init__(R,shape)
+
 class ParallelMesh(mirror.Mirror,mirrorclass=_flyft.ParallelMesh):
-    def __init__(self, L, shape, communicator):
-        super().__init__(Mesh(L,shape), communicator)
+    def __init__(self,mesh):
+        communicator = Communicator()
+        super().__init__(mesh)
```

Comment:
You shouldn't have removed the communicator here, that might be causing your error:
```suggestion
        super().__init__(mesh, communicator)
```